pkg/definition/go_gen.go

```diff
@@ -0,0 +1,290 @@
+			}
+		}
+		for i := 0; i < arguments.Len(); i++ {
+			var subParam Field
+			fi := arguments.Field(i)
+			if fi.IsDefinition {
+				continue
+			}
+			val := fi.Value
+			name := fi.Name
+			subParam.Name = name
+			switch val.IncompleteKind() {
+			case cue.StructKind:
+				if subField, err := val.Struct(); err == nil && subField.Len() == 0 { // err cannot be not nil,so ignore it
+					if mapValue, ok := val.Elem(); ok {
+						// In the future we could recursively call to support complex map-value(struct or list)
+						subParam.GoType = fmt.Sprintf("map[string]%s", mapValue.IncompleteKind().String())
+					} else {
+						// element in struct not defined, use interface{}
+						subParam.GoType = "map[string]interface{}"
+					}
+				} else {
+					if err := parseParameters(val, name); err != nil {
+						return err
+					}
+					subParam.GoType = dm.FieldName(name)
+				}
+			case cue.ListKind:
+				elem, success := val.Elem()
+				if !success {
+					// fail to get elements, use the value of ListKind to be the type
+					subParam.GoType = val.IncompleteKind().String()
+					break
+				}
+				switch elem.Kind() {
+				case cue.StructKind:
+					subParam.GoType = fmt.Sprintf("[]%s", dm.FieldName(name))
+					if err := parseParameters(elem, name); err != nil {
+						return err
+					}
+				default:
+					subParam.GoType = fmt.Sprintf("[]%s", elem.IncompleteKind().String())
+				}
+			default:
+				subParam.GoType = val.IncompleteKind().String()
+			}
+			param.Fields = append(param.Fields, Field{
+				Name:   subParam.Name,
+				GoType: subParam.GoType,
+			})
+		}
+	}
+	structs = append(structs, param)
+	return nil
+}
+
+// GenGoCodeFromParams generates go code from parameters
+func GenGoCodeFromParams(parameters []StructParameter) (string, error) {
+	var buf bytes.Buffer
+
+	for _, parameter := range parameters {
+		if parameter.Usage == "" {
+			parameter.Usage = "-"
+		}
+		fmt.Fprintf(&buf, "// %s %s\n", dm.FieldName(parameter.Name), parameter.Usage)
+		genField(parameter, &buf)
+	}
+	source, err := format.Source(buf.Bytes())
+	if err != nil {
+		fmt.Println("Failed to format source:", err)
+	}
+
+	return string(source), nil
+}
+
+// PrintParamGosStruct prints the StructParameter in Golang struct format
+func PrintParamGosStruct(parameters []StructParameter) {
+	code, err := GenGoCodeFromParams(parameters)
+	if err != nil {
+		fmt.Println("Fail to gen code, err:", err)
+	}
+	fmt.Print(code)
+}
+
+func genField(param StructParameter, buffer *bytes.Buffer) {
+	fieldName := dm.FieldName(param.Name)
+	if param.Type == cue.StructKind { // only struct kind will be separated struct
+		// cue struct  can be Go map or struct
+		if strings.HasPrefix(param.GoType, "map[string]") {
+			fmt.Fprintf(buffer, "type %s %s", fieldName, param.GoType)
+		} else {
+			fmt.Fprintf(buffer, "type %s struct {\n", fieldName)
+			for _, f := range param.Fields {
+				fmt.Fprintf(buffer, "    %s %s `json:\"%s\"`\n", dm.FieldName(f.Name), f.GoType, f.Name)
```

Comment:
Add the `omitempty` JSON tag for the field that is not required.

Reply:
done that